test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthFunctionalTestFixture.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.Dnx.CommonTestUtils;
+
+namespace Microsoft.Dnx.DesignTimeHost.FunctionalTests.Infrastructure
+{
+    public class DthFunctionalTestFixture : DnxRuntimeFixture
+    {
+        private readonly DisposableDir _context;
+        private readonly int[] _ports;
+
+        private int _nextPortIdx;
+
+        public DthFunctionalTestFixture() : base()
+        {
+            _ports = Enumerable.Range(1774, 20).ToArray();
```

Comment:
Even after we make this thread-safe, it is not robust enough. What if we have both dnx/dev and dnx/release building on the same CI machine? The port conflict still exist. In worse case, TeamCity occupy a port 1774 and this test case always fails.

We should at least add some retry logic here. I would suggest use http://stackoverflow.com/a/150974/1560415 plus retry to make it robust enough.